docs/docs/migrate-from-v1/README.md

```diff
@@ -0,0 +1,259 @@
+
+## Node.js Code Step Changes
+
+There are a few changes to the Node.js code steps that you should know about. Some functions have been renamed for more clarity, and we've aligned the Node.js code steps closer to the [Component API](/docs/components/).
+
+### Code Scaffolding Format
+
+In v1, the Node.js steps would automatically scaffold new Node.js steps in this format:
+
+```javascript
+async (event, steps) {
+  // your code could be entered in here
+}
+```
+
+In v2, the new scaffolding is wrapped with a new `defineComponent` function:
+
+```javascript
+defineComponent({
+  async run({ steps, $ }) {
+    // your code can be entered here
+  }
+});
+```
+
+1. The `event` from the trigger step is still available, but within `steps.trigger.event` instead.
+2. The `$` variable has been passed into the `run` function where your code is executed.
+
+### using 3rd party packages
+
+In v1, you had to define your imports of 3rd party packages within the scaffolded function:
+
+```javascript
+async (event, steps) {
+  const axios = require('axios');
+  // your code could be entered in here
+}
+```
+
+Now, in v2 workflows you can `import` your packages in the top of the step, just like a normal Node.js module:
+
+```javascript
+import axios from 'axios';
+
+defineComponent({
+  async run({ steps, $ }) {
+    // your code can be entered here
+  }
+});
+```
+
+Allowing all of the scaffolding to be edited opens up the ability to [pass props](code/nodejs/#passing-props-to-code-steps) into your Node.js code steps, which we'll cover later.
+
+### Step Exports 
+
+In v1, you could assign arbitrary properties to `this` within a Node.js step and the properties would be available as step exports:
+
+```javascript
+// this step's name is get_customer_data
+async (event, steps) {
+  this.name = 'Dylan';
+  // downstream steps could use steps.get_customer_data.name to retrieve 'Dylan'
+}
+```
+
+In v2, this specific way of exporting data is now available with `$.export`:
+
+```javascript
+// this step's name is get_customer_data
+defineComponent({
+  async run({ steps, $ }) {
+    $.export('name', 'Dylan');
+    // downstream steps can use steps.get_customer_data.name to retrieve 'Dylan'
+  }
+});
+```
+::: tip
+Using `return` to export data is the same from v1 to v2. You can still `return` data, and it will be available to other steps with `steps.[stepName].$return_value.
+:::
+
+### Exiting a workflow early 
+
+In v1, the `$exit` function can be called to exit a flow early:
+
+```javascript
+async (event, steps) {
+  exit('Exiting the whole workflow early');
+  console.log('I will never run');
+}
+```
+
+In v2, this same function is available, but under `$.exit`:
+
+```javascript
+defineComponent({
+  async run({ steps, $ }) {
+    $.flow.exit('Exiting the workflow early');
+    console.log('I will never run');
+  }
+});
+```
+
+::: warning
+In v2, `$end` does not display the reason in the workflow results logs.
```

Comment:
There are a few outstanding things like this, but since this should be fixed soon, I would remove this.